Tribler/Core/Modules/restapi/my_channel_endpoint.py

```diff
@@ -113,6 +118,52 @@ def render_PUT(self, request):
         return json.dumps({"added": channel_id})
 
 
+class MyChannelCreateTorrentEndpoint(MyChannelBaseEndpoint):
+    """
+    Create a torrent file from local files.
+
+    Example request:
+    {
+        "files": ['path/to/file.txt', 'path/to/another/file.mp4'],
+        "tracker": "http://localhost/announce",
+        "name": "Cat video",
+        "description: "Video of my cat"
+    }
+    """
+
+    def render_GET(self, request):
+        """
+        This method creates a torrent from local files and returns it.
+        Note that the source files need to exist and are skipped otherwise.
+        """
+        parameters = http.parse_qs(request.content.read(), 1)
+        tdef = TorrentDef()
+
+        if 'files' in parameters and len(parameters['files']) > 0:
+            for source_file in parameters['files']:
+                if os.path.exists(source_file):
+                    tdef.add_content(source_file)
+        else:
+            request.setResponseCode(http.BAD_REQUEST)
+            return json.dumps({"error": "files parameter missing"})
+
+        if 'tracker' in parameters and len(parameters['tracker']) > 0:
+            tdef.set_tracker(parameters['tracker'][0])
+        else:
+            request.setResponseCode(http.BAD_REQUEST)
+            return json.dumps({"error": "tracker parameter missing"})
+
+        if 'name' in parameters and len(parameters['name']) > 0:
+            tdef.set_name(parameters['name'][0])
+
+        if 'description' in parameters and len(parameters['description']) > 0:
+            tdef.set_comment(['description'][0])
+
+        torrent_finalized = tdef.encode()
+        torrent_64 = base64.b64encode(torrent_finalized)
+        return json.dumps({u"torrent": torrent_64})
```

Comment:
No, it is a stand-alone endpoint. You can upload this .torrent to the add-torrent endpoint right now.